src/main/java/by/savich/project/springboot/service/IphoneService.java

```diff
@@ -35,6 +35,9 @@ public void saveIphone(Iphone iphone) {
     }
 
     public Iphone getIphoneById(int id) {
+        if (iphoneRepository.findById(id) == null) {
```

Comment:
don't do twice the same: `findById()`

if `findById()` returns `Optional`, it never can be `null`
```java
        String notFoundMsg = String.format(
                "Can't find order orderId=%s for errorResponseMessage=%s", 
                orderId, toLogJson(response)
        );
        OrderEntity order = ordersRepository.findById(orderId)
                .orElseThrow(() -> new ServiceException(notFoundMsg));
```

Reply:
thanks a lot for the advice